libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -76,16 +76,15 @@ internal class MapboxRerouteController(
             }
 
             val avoidManeuverRadius = rerouteOptions.avoidManeuverSeconds.let { seconds ->
-                if (seconds == 0) {
-                    0
-                } else {
-                    (speed / seconds).toInt()
+                when (seconds) {
+                    0 -> 0
+                    else -> (speed / seconds).toInt()
                 }
             }.let { meters ->
-                if (meters > MAX_DANGEROUS_MANEUVERS_RADIUS) {
-                    MAX_DANGEROUS_MANEUVERS_RADIUS
-                } else {
-                    meters
+                when {
+                    meters == 0 -> null // passing `null` since `0` is not a valid value
```

Comment:
Should we check that it is >= 1 according to the error message? 
Just in case we ended up with a negative value. 
We can also do something like this:
```kotlin
val avoidManeuverRadius = rerouteOptions.avoidManeuverSeconds
    .takeIf { it != 0 }
    ?.let { speed / it }
    ?.toInt()
    ?.takeIf { it >= 1 }
    ?.coerceAtMost(MAX_DANGEROUS_MANEUVERS_RADIUS)
```

Reply:
It's an int so if it's not zero, it will meet the condition. I do like your suggestion though, it tells the story better :+1: